### ISSUE TITLE
Building new version of polybar in openSUSE 15.2

### ISSUE THREAD

**Issue by sysek** [polybar/polybar#2316]
## Checklist

* [x] I have followed every step on the [compiling wiki page](https://github.com/polybar/polybar/wiki/Compiling) and
  installed all necessary dependencies.
* [x] My problem is not described on the [known issues page](https://github.com/polybar/polybar/wiki/Known-Issues)
* [x] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that
  may have already reported this problem.
* [x] I was able to reproduce this build issue in a clean build

## Build Process

**How did you build polybar?**

* [x] From a release archive (Which version?) - 3.5.2
* [ ] By cloning this repository (What is the output of `git describe --tags`?)
* [ ] Some other way (How?)

## Build Log

```
[    5s] make: Entering directory '/home/abuild/rpmbuild/BUILD/polybar-3.5.2/build'
[    5s] /usr/bin/cmake -S/home/abuild/rpmbuild/BUILD/polybar-3.5.2 -B/home/abuild/rpmbuild/BUILD/polybar-3.5.2/build --check-build-system CMakeFiles/Makefile.cmake 0
[    5s] /usr/bin/cmake -E cmake_progress_start /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build/CMakeFiles /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build/CMakeFiles/progress.marks
[    5s] /usr/bin/make -f CMakeFiles/Makefile2 all
[    5s] make[1]: Entering directory '/home/abuild/rpmbuild/BUILD/polybar-3.5.2/build'
[    5s] /usr/bin/make -f doc/CMakeFiles/doc_html.dir/build.make doc/CMakeFiles/doc_html.dir/depend
[    5s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/polybar-3.5.2/build'
[    5s] cd /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/abuild/rpmbuild/BUILD/polybar-3.5.2 /home/abuild/rpmbuild/BUILD/polybar-3.5.2/doc /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build/doc /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build/doc/CMakeFiles/doc_html.dir/DependInfo.cmake
[    5s] Scanning dependencies of target doc_html
[    5s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/polybar-3.5.2/build'
[    5s] /usr/bin/make -f doc/CMakeFiles/doc_html.dir/build.make doc/CMakeFiles/doc_html.dir/build
[    5s] make[2]: Entering directory '/home/abuild/rpmbuild/BUILD/polybar-3.5.2/build'
[    5s] [  1%] sphinx-build html: see doc/builder-html.log
[    5s] cd /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build/doc && sphinx-build -b html -c /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build/doc -d /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build/doc/doctrees /home/abuild/rpmbuild/BUILD/polybar-3.5.2/doc /home/abuild/rpmbuild/BUILD/polybar-3.5.2/build/doc/html > builder-html.log
[    6s] Traceback (most recent call last):
[    6s]   File "/usr/lib/python3.6/site-packages/sphinx/config.py", line 161, in __init__
[    6s]     execfile_(filename, config)
[    6s]   File "/usr/lib/python3.6/site-packages/sphinx/util/pycompat.py", line 150, in execfile_
[    6s]     exec_(code, _globals)
[    6s]   File "conf.py", line 20, in <module>
[    6s]     from sphinx.domains.changeset import VersionChange
[    6s] ModuleNotFoundError: No module named 'sphinx.domains.changeset'
[    6s] 
[    6s] make[2]: *** [doc/CMakeFiles/doc_html.dir/build.make:78: doc/CMakeFiles/doc_html] Error 2
[    6s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/polybar-3.5.2/build'
[    6s] make[1]: *** [CMakeFiles/Makefile2:515: doc/CMakeFiles/doc_html.dir/all] Error 2
[    6s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/polybar-3.5.2/build'
[    6s] make: *** [Makefile:150: all] Error 2
[    6s] make: Leaving directory '/home/abuild/rpmbuild/BUILD/polybar-3.5.2/build'
[    6s] error: Bad exit status from /var/tmp/rpm-tmp.kH66yQ (%install)
[    6s] 
```

## Environment:
* Distro (with Version): openSUSE 15.2

## Additional context

Hi, 

I tried to update polybar in openSUSE 15.2 but get issues with compiling. As you see above there's a problem with Sphinx, in openSUSE 15.2 the version of that package is 1.7.6 (maybe that is the issue?). What is the minimum version required to pass build?

**Comment by patrick96** [comment]
I actually can't find this module anywhere in the sphinx documentation. But looking at the repo, it was first released in version 1.8.

The (very) old sphinx version is definitely at fault here. Does openSUSE really have now newer version of this package?

**Comment by sysek** [comment]
> Does openSUSE really have now newer version of this package?

uhm, about that.. yes, the Leap version have 1.7.6 version in repo, newer version is on Tumbleweed (3.3.1).

Anyway, thanks for reply I assume, we can close that ticket as resolved, because the older version is the problem.

**Comment by patrick96** [comment]
Oh that's too bad. But I think in this case installing a newer version through pip also works.

If you are talking from a packaging standpoint, you could either disable the documentation by passing `-DBUILD_DOC=OFF` to `cmake` (this also disable the manpages) or remove the offending code from the `conf.py`.

Patching the `conf.py` would involve removing the import `from sphinx.domains.changeset import VersionChange` and removing the last 23 lines (everything after `suppress_warnings = ['app.add_directive']`).

**Comment by patrick96** [comment]
Oh, I totally forgot about this. There was another issue with the same problem and I fixed this for version 3.5.5, so you should be able to build polybar again even on opensuse leap
